src/adapters/adapter.gitlab.js

```diff
@@ -0,0 +1,282 @@
+  , GL_BRANCH_SEL  = '[aria-label="Switch branches or tags"]'
+  , GL_404_SEL          = '#parallax_wrapper'
+  , GL_PJAX_SEL         = '#js-repo-pjax-container'
+  , GL_CONTAINERS       = 'body > .page-with-sidebar'
+
+function GitLab() {
+  if (!window.MutationObserver) return
+
+  // Fix #151 by detecting when page layout is updated.
+  // In this case, split-diff page has a wider layout, so need to recompute margin.
+  // Note that couldn't do this in response to URL change, since new DOM via pjax might not be ready.
+  var observer = new window.MutationObserver(function(mutations) {
+    for (var i = 0, len = mutations.length; i < len; i++) {
+      var mutation = mutations[i]
+      if (~mutation.oldValue.indexOf('split-diff') ||
+          ~mutation.target.className.indexOf('split-diff')) {
+        return $(document).trigger(EVENT.LAYOUT_CHANGE)
+      }
+    }
+  })
+
+  observer.observe(document.body, {
+    attributes: true,
+    attributeFilter: ['class'],
+    attributeOldValue: true
+  })
+}
+
+/**
+ * Selects a submodule.
+ */
+GitLab.prototype.selectSubmodule = function(path) {
+  window.location.href = path
+}
+
+/**
+ * Selects a path.
+ */
+GitLab.prototype.selectPath = function(path, tabSize) {
+  var container = $(GL_PJAX_SEL)
+    , qs = tabSize ? ('?ts=' + tabSize) : ''
+
+  if (container.length) {
+    $.pjax({
+      // needs full path for pjax to work with Firefox as per cross-domain-content setting
+      url : location.protocol + '//' + location.host + path + qs,
+      container : container
+    })
+  }
+  else window.location.href = path + qs // falls back if no container (i.e. GitHub DOM has changed or is not yet available)
+}
+
+/**
+ * Append sidebar to corresponding place
+ */
+GitLab.prototype.appendSidebar = function(sidebar) {
+  sidebar
+    .insertAfter($('.sidebar-wrapper'))
+    .addClass('octotree_gitlab_sidebar')
+}
+
+/**
+ * Updates page layout based on visibility status and width of the Octotree sidebar.
+ */
+GitLab.prototype.updateLayout = function(sidebarVisible, sidebarWidth) {
+  var $containers = $(GL_CONTAINERS)
+    , spacing = 10
+    , autoMarginLeft
+    , shouldPushLeft
+  if ($containers.length === 1) {
+    autoMarginLeft = ($('body').width() - $containers.width()) / 2
+    shouldPushLeft = sidebarVisible && (autoMarginLeft <= sidebarWidth + spacing)
+    $containers.css('margin-left', shouldPushLeft ? sidebarWidth + spacing : '')
+  }
+
+  // falls-back if GitHub DOM has been updated
+  else $('html').css('margin-left', sidebarVisible ? sidebarWidth - spacing : '')
+}
+
+/**
+ * Returns the repository information if user is at a repository URL. Returns `null` otherwise.
+ */
+GitLab.prototype.getRepoFromPath = function(showInNonCodePage, currentRepo) {
+  // 404 page, skip
+  if ($(GL_404_SEL).length) return false
+
+  // (username)/(reponame)[/(type)]
+  var match = window.location.pathname.match(/([^\/]+)\/([^\/]+)(?:\/([^\/]+))?/)
+  if (!match) return false
```

Comment:
Expected '{' and instead saw 'return'.<br>Missing semicolon.<br>Too many errors. (34% scanned).